skills_taxonomy_v2/pipeline/skills_extraction/skills_naming_utils.py

```diff
@@ -148,67 +220,93 @@ def get_clean_ngrams(sentence_skills, ngram, min_count, threshold):
+    ]
 
+    # lemmatise ngrams
+    clean_ngrams = [
+        " ".join([lemmatizer.lemmatize(n) for n in ngram.split(" ")])
+        for ngram in clean_ngrams
+    ]
 
-def get_skill_info(
-    clean_ngrams, sentence_skills, sentence_embs, cluster_descriptions, num_top_sent=2
-):
+    # only return ngrams that are more than 1 word long
+    return (
+        [clean for clean in clean_ngrams if len(clean.split(" ")) > 1],
+        cluster_descriptions,
+    )
+
+
+def get_skill_info(skills_df, num_top_sent, ngram, min_count, threshold):
     """
+    Inputs:
+        'skills_df' (dataframe)
+    
     Output: skills_data (dict), for each skill number:
-        'Skills name' : join the closest ngram to the centroid
+        'Skills name' : the closest ngram to the centroid or, if no ngrams generated, shortest description
         'Examples': Join the num_top_sent closest original sentences to the centroid
         'Texts': All the cleaned sentences for this cluster
     """
 
     bert_vectorizer = BertVectorizer(
-        bert_model_name="sentence-transformers/paraphrase-MiniLM-L6-v2",
-        multi_process=True,
+        bert_model_name="sentence-transformers/all-MiniLM-L6-v2", multi_process=True,
     )
     bert_vectorizer.fit()
 
-    # embed ngrams
-    ngram_embeddings = bert_vectorizer.transform(clean_ngrams)
-
-    # calculate similarities
-    skills_data = {}
-    for cluster_num, cluster_data in tqdm(sentence_skills.groupby("Cluster number")):
-        # There may be the same sentence repeated
-        cluster_data.drop_duplicates(["sentence id"], inplace=True)
-        cluster_text = cluster_data["original sentence"].tolist()
-        cluster_coords = cluster_data[["reduced_points x", "reduced_points y"]].values
-        cluster_embeds = [
-            np.array(sentence_embs[str(sent_id)]).astype("float32")
-            for sent_id in cluster_data["sentence id"].values.tolist()
-            if str(sent_id) in sentence_embs
+    skill_data = {}
+    for skills_num, skills_data in skills_df.iterrows():
+        sents = skills_data["Sentences"]
+        reduced_centroid_embeds = np.array(skills_data["Centroid"]).astype("float32")
+        centroid_embeds = np.array(skills_data["Mean embedding"]).astype("float32")
+        sent_embeds = [
+            np.array(sent_embed).astype("float32")
+            for sent_embed in skills_data["Sentence embeddings"]
         ]
 
-        # Get sent similarities to centre
         sent_similarities = cosine_similarity(
-            np.mean(cluster_coords, axis=0).reshape(1, -1), cluster_coords
+            reduced_centroid_embeds.reshape(1, -1), sent_embeds
         )
 
-        # calculate similarities between ngrams per cluster and cluster mean
-        ngram_similarities = cosine_similarity(
-            np.mean(cluster_embeds, axis=0).reshape(1, -1), ngram_embeddings
+        candidate_ngrams, cluster_descriptions = get_clean_ngrams(
+            sents, skills_num, ngram, min_count, threshold
```

Comment:
I think if you remove `skills_num` as an input to `get_clean_ngrams` then the output `cluster_descriptions` can just be a list rather than a one key dictionary (so later you can do `"Texts": cluster_descriptions` rather than `"Texts": cluster_descriptions[skills_num]`)